cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/AlertaSelfHealingNotifier.java

```diff
@@ -0,0 +1,261 @@
+        }
+        final String localHostname = tmpLocalHostname;
+
+        List<AlertaMessage> alertaMessages = new ArrayList<>();
+
+        switch ((KafkaAnomalyType) anomalyType) {
+          case GOAL_VIOLATION:
+            GoalViolations goalViolations = (GoalViolations) anomaly;
+            alertGoalViolation(anomalyType, localHostname, alertaMessages, goalViolations);
+            break;
+          case BROKER_FAILURE:
+            BrokerFailures brokerFailures = (BrokerFailures) anomaly;
+            alertBrokerFailure(anomalyType, localHostname, alertaMessages, brokerFailures);
+            break;
+          case METRIC_ANOMALY:
+            KafkaMetricAnomaly metricAnomaly = (KafkaMetricAnomaly) anomaly;
+            alertMetricAnomaly(anomalyType, localHostname, alertaMessages, metricAnomaly);
+            break;
+          case DISK_FAILURE:
+            DiskFailures diskFailures = (DiskFailures) anomaly;
+            alertDiskFailure(anomalyType, localHostname, alertaMessages, diskFailures);
+            break;
+          case TOPIC_ANOMALY:
+            TopicAnomaly topicAnomaly = (TopicAnomaly) anomaly;
+            alertTopicAnomaly(anomalyType, localHostname, alertaMessages, topicAnomaly);
+            break;
+          case MAINTENANCE_EVENT:
+            MaintenanceEvent maintenanceEvent = (MaintenanceEvent) anomaly;
+            alertMaintenanceEvent(anomalyType, localHostname, alertaMessages, maintenanceEvent);
+            break;
+          default:
+            throw new IllegalStateException("Unrecognized anomaly type.");
+        }
+
+        for (AlertaMessage alertaMessage : alertaMessages) {
+          alertaMessage.setEnvironment(_alertaEnvironment);
+          alertaMessage.setService(Collections.singletonList(ALERT_CRUISE_CONTROL));
+          alertaMessage.setText(text);
+          alertaMessage.setOrigin(ALERT_CRUISE_CONTROL + "/" + localHostname);
+          alertaMessage.setType(ALERT_CRUISE_CONTROL_ALARM);
+          alertaMessage.setRawData(anomaly.toString());
+          alertaMessage.setTags(Collections.singletonList(ALERT_ALARM_ID_TAG_KEY + ":" + anomaly.anomalyId()));
+          try {
+              sendAlertaMessage(alertaMessage);
+          } catch (IOException e) {
+              LOG.warn("ERROR sending alert to Alerta.io", e);
+          }
+        }
+    }
+
+    private void alertMaintenanceEvent(AnomalyType anomalyType, final String localHostname,
+            List<AlertaMessage> alertaMessages, MaintenanceEvent maintenanceEvent) {
+        AlertaMessage alertaMessage = new AlertaMessage(localHostname, anomalyType.toString() + " - " + maintenanceEvent.reasonSupplier());
+        alertaMessage.setSeverity(NotifierUtils.getAlertSeverity(anomalyType).toString());
+        alertaMessage.setGroup(AlertaAlertGroup.PERFORMANCE.toString());
+        alertaMessage.setCreateTime(CruiseControlUtils.utcDateFor(maintenanceEvent.detectionTimeMs(), 3, ChronoUnit.SECONDS));
```

Comment:
Sorry for a question on old commit. 
I'm reading related implementation and I wondered why you need  precision for milliseconds even you truncate. I understood the reason you need milliseconds from this [comment](https://github.com/linkedin/cruise-control/pull/1510#issuecomment-832978240) but why you don't set ChronoUnit.MILLISECONDS on the 3rd parameter?